File: src/domain/TransferQuote.ts
import { CurrencyPair } from './CurrencyPair.js';
import { Money } from './Money.js';

export interface TransferQuote {
  pair: CurrencyPair;
  exchangeRate: number;
  amountToSend: Money;
  amountToReceive: Money;
  fee: Money;
  totalDebit: Money;
}

File: internal/domain/transfer_quote.go
package domain

import "github.com/shopspring/decimal"

// TransferQuote encapsulates the outcome of calculating a transfer between two currencies.
type TransferQuote struct {
	Pair          CurrencyPair
	ExchangeRate  decimal.Decimal
	SendAmount    Money
	ReceiveAmount Money
	Fee           Money
	TotalDebit    Money
	Allowed       bool
}
